### PR TITLE
Utilise larger GitHub hosted runners

### PR DESCRIPTION
**Pull request by fcooper8472**
See #232

### CODE REVIEW

File: .github/workflows/docker-develop-image.yml
Comment (fcooper8472) [inline .github/workflows/docker-develop-image.yml:12]: @bdevans does this look appropriate? There is a dedicated runner tagged with this label, but are there other changes likely to be needed? I'm not sure we can currently test this workflow without merging.

File: .github/workflows/docker-release-image.yml
Comment (fcooper8472) [inline .github/workflows/docker-release-image.yml:16]: @bdevans does this look appropriate? There is a dedicated runner tagged with this label, but are there other changes likely to be needed? I'm not sure we can currently test this workflow without merging.

File: .github/workflows/docker-develop-image.yml
Comment (bdevans) [reply .github/workflows/docker-develop-image.yml:12]: Yes, assuming you have tagged the machine with `self-hosted-docker`(before it was just `docker`). 
Thanks, once that's merged it should start working again...! 🤞

**Comment by fcooper8472** [comment]
I've put in a temporary fix for coverage in the random field generation code; see #234 for a longer-term solution.

File: .github/workflows/docker-develop-image.yml
Comment (bdevans) [reply .github/workflows/docker-develop-image.yml:12]: Actually, if you tagged it as `docker` then this would mean we could rerun the original job to build the `chaste/release` image and test that it works. The tag could then be changed to the more descriptive `self-hosted-docker` afterwards.

**Comment by fcooper8472** [comment]
f23ec4a2a47032e3db5393e291a343198aea5d42 adds some statistically appropriate tests for random field generation. This is enough to sensibly fix the coverage error, but further work is needed. See #234.

**Comment by fcooper8472** [comment]
Pending green ticks, this should be ready for review.

@bdevans or @kwabenantim, could you review with a mind to the GitHub runner changes?

@AlexFletcher @mirams or @jmpf, could one of you take a quick look at the testing changes to random fields? They're related to #234, but everything is blocked until we have a green tick on coverage, so the minimum work on that has been added here.

**Review by bdevans — APPROVED** [review]
👍

**Review by kwabenantim — COMMENTED** [review]
The workflow changes look good to me.

**Comment by fcooper8472** [comment]
The coverage fix does not need to be mentioned in changes since last release, as it just fixes two lines that were uncovered at the point of the last release.

**Comment by fcooper8472** [comment]
@jmpf approves.

**Pull request merged**
